source/common/config/grpc_mux_impl.cc

```diff
@@ -84,17 +67,17 @@ void GrpcMuxImpl::queueDiscoveryRequest(const std::string& type_url) {
   drainRequests();
 }
 
-void GrpcMuxImpl::sendDiscoveryRequest(const std::string& type_url) {
-  if (stream_ == nullptr) {
+bool GrpcMuxImpl::sendDiscoveryRequest(const std::string& type_url) {
+  if (!discovery_grpc_stream_.available()) {
     ENVOY_LOG(debug, "No stream available to sendDiscoveryRequest for {}", type_url);
-    return;
+    return true; // TODO(fredlas) 'true' is the original behavior; should it be changed?
```

Comment:
Having this case return true rather than false sticks to the existing behavior; the unit tests pass. However, it seems to me that the existing behavior is incorrect: a request that is attempted to be sent during a hiccup will be silently dropped, rather than left in the queue. I wanted to make sure I'm not misunderstanding what is intended here before I dive into changing all the unit tests to expect this behavior.

Reply:
See other comment, would rather keep the refactor and behavioral change PR separate.